components/match2/wikis/teamfortress/match_summary.lua

```diff
@@ -0,0 +1,157 @@
+---
+-- @Liquipedia
+-- wiki=teamfortress
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local Icon = require('Module:Icon')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Page = require('Module:Page')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper')
+local MatchSummary = Lua.import('Module:MatchSummary/Base')
+
+local htmlCreate = mw.html.create
+
+---@enum TFMatchIcons
+local Icons = {
+	CHECK = Icon.makeIcon{iconName = 'winner', color = 'forest-green-text', size = 'initial'},
+	EMPTY = '[[File:NoCheck.png|link=]]',
+}
+
+local LINK_DATA = {
+	logstf = {icon = 'File:Logstf_icon.png', text = 'logs.tf Match Page '},
+	logstfgold = {icon = 'File:Logstf_gold_icon.png', text = 'logs.tf Match Page (Golden Cap) '},
+	esl = {
+		icon = 'File:ESL 2019 icon lightmode.png',
+		iconDark = 'File:ESL 2019 icon darkmode.png',
+		text = 'ESL matchpage'
+	},
+	esea = {icon = 'File:ESEA icon allmode.png', text = 'ESEA Match Page'},
+	etf2l = {icon = 'File:ETF2L.png', text = 'ETF2L Match Page'},
+	rgl = {icon = 'File:RGL_Logo.png', text = 'RGL Match Page'},
+	ozf = {icon = 'File:ozfortress-icon.png‎', text = 'ozfortress Match Page'},
+	tftv = {icon = 'File:Teamfortress.tv.png', text = 'TFTV Match Page'},
+}
+
+local CustomMatchSummary = {}
+
+---@param args table
+---@return Html
+function CustomMatchSummary.getByMatchId(args)
+	return MatchSummary.defaultGetByMatchId(CustomMatchSummary, args)
+end
+
+---@param match MatchGroupUtilMatch
+---@param footer MatchSummaryFooter
+---@return MatchSummaryFooter
+function CustomMatchSummary.addToFooter(match, footer)
+	footer = MatchSummary.addVodsToFooter(match, footer)
+		:addLinks(LINK_DATA, match.links)
+
+	return footer
+end
+
+---@param match MatchGroupUtilMatch
+---@return MatchSummaryBody
+function CustomMatchSummary.createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or DateExt.isDefaultTimestamp(match.timestamp) then
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not default timestamp, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	Array.forEach(match.games, function(game)
+		if not game.map then return end
```

Comment:
Why is this needed?

Reply:
If this line is removed, the summary no longer shows the per map result
![image](https://github.com/user-attachments/assets/b652804f-7dbd-4dc3-aede-b1b929d65119)
![image](https://github.com/user-attachments/assets/5bbb19a3-e582-452e-a196-9e41f11c82b5)

Reply:
Asking about L75, the rest is obviously needed